### PR TITLE
Docs: Fundamentals of Block Development - Working with Javascript in the Block Editor

### PR DESCRIPTION
**Pull request by juanmaguitar**
> [!NOTE]  
> I opened this new PR to avoid some unwanted commits I got on the original one https://github.com/WordPress/gutenberg/pull/56294 when trying to get that branch rebased from trunk

Add documentation for working with Javascript in the Block Editor
This PR closes the issue https://github.com/WordPress/gutenberg/issues/56277

This guide along with the rest of the docs planned for https://github.com/WordPress/gutenberg/issues/56282 will allow to remove the How-To guide: [How to use JavaScript with the Block Editor](https://developer.wordpress.org/block-editor/how-to-guides/javascript/)

### CODE REVIEW

**Comment by juanmaguitar** [comment]
@fabiankaegy as you reviewed https://github.com/WordPress/gutenberg/pull/56294, can you have a look at this PR and approve it if everything is OK?

I'll merge it once I get the proper approvals without adding it to the sidebar.
I'll add the pages available for  the "Fundamentals of Block Development" to the sidebar (add them to the manifesto) with a PR for https://github.com/WordPress/gutenberg/issues/56282 once a few of the pages described on https://github.com/WordPress/gutenberg/issues/56282 have been merged

File: docs/getting-started/fundamentals-block-development/javascript-in-the-block-editor.md
Comment (fabiankaegy) [inline docs/getting-started/fundamentals-block-development/javascript-in-the-block-editor.md]:
```suggestion
Without a build process, you access the methods directly from the `wp` global object and must enqueue the script manually. [WordPress Javascript](https://developer.wordpress.org/block-editor/reference-guides/packages/) packages](https://developer.wordpress.org/block-editor/reference-guides/packages/) can be accessed through the `wp` [global variable](https://developer.mozilla.org/en-US/docs/Glossary/Global_variable) but every script that wants to use them through this `wp` object is responsible for adding [the handle of that package](https://developer.wordpress.org/block-editor/contributors/code/scripts/) to the dependency array when registered.
```

**Review by fabiankaegy — APPROVED** [review]
Found one minor improvement but approving as is :)

**Pull request merged**
